phantom/env.py

```diff
@@ -135,27 +145,26 @@ def resolve_network(self) -> None:
         self.network.resolve(self._ctxs)
         self.post_message_resolution()
 
-    def reset(self, sample_supertypes: bool = True) -> Dict[AgentID, Any]:
+    def reset(self) -> Dict[AgentID, Any]:
         """
         Reset the environment and return an initial observation.
 
         This method resets the step count and the :attr:`network`. This includes all the
         agents in the network.
 
-        Arguments:
-            sample_supertypes: If set to False will not automatically sample supertypes
-                and apply agent and environment types.
-
         Returns:
             A dictionary mapping Agent IDs to observations made by the respective
             agents. It is not required for all agents to make an initial observation.
         """
         # Reset clock
         self.current_step = 0
 
-        # Sample from supertypes and apply to agent and env type objects
-        if sample_supertypes:
-            self.sample_supertypes()
+        # Sample from supertype shared sampler objects
+        for sampler in self._samplers:
+            sampler.value = sampler.sample()
```

Comment:
can this assignment be done within the `sample` method of the sampler object? I find it strange to change the property of the object outside the scope of the object

Reply:
Fixed. Now the `.value` is a `@property` of the `Sampler` class and is set with the `.sample()` method. This is better than before as the `.value` property is now (as much as Python is) read only to users of the class